server/web/httpserver.go

```diff
@@ -1,16 +1,52 @@
 package web
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
+	gw "github.com/TUM-Dev/Campus-Backend/api"
+	"github.com/TUM-Dev/Campus-Backend/backend"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
 	"net"
 	"net/http"
 )
 
-func HTTPServe(l net.Listener) error {
-	mux := http.NewServeMux()
-	mux.HandleFunc("/", func(w http.ResponseWriter, _ *http.Request) {
-		w.Write([]byte("Welcome to the new campus API."))
-	})
+func HTTPServe(l net.Listener, grpcPort string) error {
+	mux := runtime.NewServeMux(runtime.WithIncomingHeaderMatcher(
+		func(s string) (string, bool) {
+			return s, true
+		}),
+		runtime.WithErrorHandler(errorHandler),
+	)
+	opts := []grpc.DialOption{grpc.WithInsecure()}
+	err := gw.RegisterCampusHandlerFromEndpoint(context.TODO(), mux, grpcPort, opts)
```

Comment:
> Do not pass a nil Context, even if a function permits it. Pass context.TODO if you are unsure about which Context to use.

You sure about using this?

Reply:
Went with `context.WithCancel` now :ok_hand: